coremltools/converters/nnssa/coreml/graph_pass/mlmodel_passes.py

```diff
@@ -244,66 +245,176 @@ def remove_redundant_transposes(spec):
-        list and will not find pairs in more complicated structure.
+        Search the neural network spec for sequence of transposes that together
+        are the identity, and return a list of those sequence.
         """
         nn_layers = nn_spec.layers
         layers_to_delete = []
-        # This holds the axes definition if the previous layer was a transpose,
-        # otherwise it is None.
-        previous_transpose = None
-        for _layer in nn_layers:
-            layer_type = _layer.WhichOneof('layer')
-            if layer_type != 'transpose' or len(_layer.output) != 1:
-                previous_transpose = None
-                continue
 
-            if not previous_transpose:
-                previous_transpose = {'layer': _layer, 'axes':_layer.transpose.axes}
-            else:
-                # Check if the layers are connected in the graph.
-                connected = (previous_transpose['layer'].output == _layer.input)
-
-                # Check if they're each other's inverses.
-                if connected:
-                    this_transpose = _layer.transpose.axes
-                    composed = [previous_transpose['axes'][i] for i in this_transpose]
-                    compose_to_identity = all([ax == i for i, ax in enumerate(composed)])
-
-                if connected and compose_to_identity:
-                    # These transpose ops are redundant, remove them.
-                    layers_to_delete.append((previous_transpose['layer'], _layer))
+        # Check no branch or loop in the network...
+        for layer in nn_layers:
+            layer_type = layer.WhichOneof('layer')
+            if layer_type == 'loop' or layer_type == 'branch':
+                return layers_to_delete
+
+        input_to_parent_layers, output_to_layers = blob_name_to_layers(nn_layers)
+
+        for layer in nn_layers:
+            # Only start with the last element of the transpose layers sequence
+            if not layer.WhichOneof('layer') == 'transpose':
+                continue
+            if layer.output[0] in output_to_layers and \
+               len(output_to_layers[layer.output[0]]) == 1 and \
+               output_to_layers[layer.output[0]][0].WhichOneof('layer') == 'transpose':
+               continue
+
+            # Get the transpose layers sequence
+            layers = []
+            cursor = layer
+            while True:
+                if cursor.output[0] in output_to_layers:
+                    layers.append(cursor)
+                if not cursor.input[0] in input_to_parent_layers:
+                    break
+                cursor = input_to_parent_layers[cursor.input[0]]
+                if cursor.WhichOneof('layer') != 'transpose':
+                    break
+                if len(output_to_layers[cursor.output[0]]) != 1:
+                    break
+            layers = layers[::-1]
+
+            if len(layers) == 0: continue
```

Comment:
@aseemw  I add this line for the case where the only transpose op is in outputs.